commands/utils/packagehandlers/yarnpackagehandler.go

```diff
@@ -19,5 +33,50 @@ func (yarn *YarnPackageHandler) UpdateDependency(vulnDetails *utils.Vulnerabilit
 }
 
 func (yarn *YarnPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails, extraArgs ...string) (err error) {
-	return yarn.CommonPackageHandler.UpdateDependency(vulnDetails, extraArgs...)
+	executableYarnVersion, err := getYarnVersion()
```

Comment:
we need yarn version only for the isV2andAbove so we can combine them together to one func -  isYarnV2orAbove() or isYarnV1() that gets yarn version and checks if v2 or above